### PR TITLE
Fixed the SaveKeyValue destination account check

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed the destination account check in the SaveKeyValue builtin function

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: builtInFunctions/keyValueStorage.go [hunk 1]
```diff
@@ -44,13 +44,13 @@ func (k *saveKeyValueStorage) SetNewGasConfig(gasCost *vmcommon.GasCost) {
...
 func (k *saveKeyValueStorage) ProcessBuiltinFunction(
-	acntSnd, _ vmcommon.UserAccountHandler,
+	acntSnd, acntDest vmcommon.UserAccountHandler,
 	input *vmcommon.ContractCallInput,
 ) (*vmcommon.VMOutput, error) {
 	k.mutExecution.RLock()
 	defer k.mutExecution.RUnlock()
 
-	err := checkArgumentsForSaveKeyValue(acntSnd, input)
+	err := checkArgumentsForSaveKeyValue(acntDest, input)
```
Comment (sasurobert) [inline builtInFunctions/keyValueStorage.go:53]: shouldn't we use acntDest on line 74 and on line 93 ?

**Review by sasurobert — APPROVED** [review]

File: builtInFunctions/keyValueStorage_test.go [hunk 2]
```diff
@@ -85,23 +85,23 @@ func TestSaveKeyValue_ProcessBuiltinFunction(t *testing.T) {
 	value := []byte("value")
 	vmInput.Arguments = [][]byte{key, value}
 
-	_, err = skv.ProcessBuiltinFunction(nil, nil, vmInput)
+	_, err = skv.ProcessBuiltinFunction(acc, nil, vmInput)
 	require.Equal(t, ErrNilSCDestAccount, err)
 
-	_, err = skv.ProcessBuiltinFunction(acc, nil, vmInput)
+	_, err = skv.ProcessBuiltinFunction(acc, acc, vmInput)
```
Comment (bogdan-rosianu) [inline builtInFunctions/keyValueStorage_test.go:91]: `(nil, acc)` in the entire file where the destination isn't nil?
the sender is ignored so maybe reflect this in the tests as well

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: builtInFunctions/keyValueStorage_test.go (hunk 2 above)
Comment (iulianpascalau) [reply builtInFunctions/keyValueStorage_test.go:91]: blockchain hook will never provide a nil account for sender. It is against how the protocol was implemented.

Comment (iulianpascalau) [reply builtInFunctions/keyValueStorage_test.go:91]: will add one more test for nil sender

**Review by sasurobert — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]
